### PR TITLE
Correcting the relocation according to query

### LINKED ISSUE #462: Change the URL of the query with whole query
* Currently the URL changes after the query only with the display string

* This is not the desired behavior. The url must change with the whole query including the string and the other aspects, like filter etc

* Use `routerString` to enable this functionality

### PR DESCRIPTION
**Pull request by hemantjadon**
* Update the URL according to the query fetched. 

* This allows the easy shareability of query URL's in their complete form including filters and other customizations

**Closes #462**

### CODE REVIEW

File: src/app/feed/feed.component.ts [hunk 1]
```diff
@@ -86,10 +96,14 @@ export class FeedComponent implements OnInit, AfterViewInit, OnDestroy {
 	private queryFromURL(): void {
 		this.__subscriptions__.push(
 			this.route.queryParams
-					.subscribe((params: Params) => {
-						const queryParam = params['query'] || '';
-						this.search(queryParam);
```
Comment (Achint08) [inline src/app/feed/feed.component.ts]: Remove 'search` method.

File: src/app/pagenotfound/pagenotfound.component.spec.ts [hunk 2]
```diff
@@ -1,11 +1,21 @@
 /* tslint:disable:no-unused-variable */
 
+import { Title } from '@angular/platform-browser';
 import { TestBed, async } from '@angular/core/testing';
 import { PageNotFoundComponent } from './pagenotfound.component';
 
+
 describe('Component: Terms', () => {
```
Comment (Achint08) [inline src/app/pagenotfound/pagenotfound.component.spec.ts]: It should be `Component: PageNotFound`

File: src/app/pagenotfound/pagenotfound.component.spec.ts [hunk 3]
```diff
@@ -1,11 +1,21 @@
...
+	beforeEach(() => {
+		TestBed.configureTestingModule({
```
Comment (Achint08) [inline src/app/pagenotfound/pagenotfound.component.spec.ts]: Add `PageNotFoundComponent` to the declaration.

File: src/app/feed/feed.component.ts (hunk 1 above)
Comment (hemantjadon) [inline src/app/feed/feed.component.ts]: @Achint08 this is already removed. See it is in red section

Comment (Achint08) [inline src/app/feed/feed.component.ts]: I was talking about
https://github.com/hemantjadon/loklak_search/blob/232fdf43e6462433eafa09dca21225a0231cd2c3/src/app/feed/feed.component.ts#L139

Comment (hemantjadon) [inline src/app/feed/feed.component.ts]: No you can't remove that it is still getting called on query change from the header.

**Comment by Achint08** [comment]
@hemantjadon Please Solve Merge conflicts. :)

**Comment by hemantjadon** [comment]
@Achint08 @singhpratyush @vibhcool @kavithaenair @djmgit Please review

**Pull request merged**